src/runtime_shared.js

```diff
@@ -61,6 +61,53 @@ var wasmOffsetConverter;
 #include "wasm_offset_converter.js"
 #endif
 
+// Memory management
+
+#if !WASM_ESM_INTEGRATION || IMPORTED_MEMORY
+var wasmMemory;
+#endif
+
+var
+/** @type {!Int8Array} */
+  HEAP8,
+/** @type {!Uint8Array} */
+  HEAPU8,
+/** @type {!Int16Array} */
+  HEAP16,
+/** @type {!Uint16Array} */
+  HEAPU16,
+/** @type {!Int32Array} */
+  HEAP32,
+/** @type {!Uint32Array} */
+  HEAPU32,
+/** @type {!Float32Array} */
+  HEAPF32,
+/** @type {!Float64Array} */
+  HEAPF64;
+
+#if WASM_BIGINT
+// BigInt64Array type is not correctly defined in closure
+var
+/** not-@type {!BigInt64Array} */
+  HEAP64,
+/* BigUint64Array type is not correctly defined in closure
+/** not-@type {!BigUint64Array} */
+  HEAPU64;
+#endif
+
+#if SUPPORT_BIG_ENDIAN
+/** @type {!DataView} */
+var HEAP_DATA_VIEW;
+#endif
+
+#if !MINIMAL_RUNTIME || ASSERTIONS || SAFE_HEAP || USE_ASAN
+var runtimeInitialized = false;
+#endif
```

Comment:
Maybe leave this in the non-minimal file rather than moving it and also adding `!MINIMAL_RUNTIME`

Reply:
Perhaps put it right along side the `initRuntime` function that uses it?

Reply:
> Perhaps put it right along side the `initRuntime` function that uses it?

You mean in both preamble and preamble_minimal?

Reply:
It doesn't need to go in preamble_minimal, right?  Since this is not a variable that exists under `-sMINIMUMAL_RUNTIME`

Reply:
It does if those other conditions are met. It existed in both files, I just merged the conditions when extracting.

Reply:
Oops I misunderstood.